File: ruby/lib/command/file_handler/decompress.rb
module Command
  module FileHandler
    class Decompress < Base

      def load
        if valid_input_file?
          File.binread(input_path)
        else
          raise 'Invalid File'
        end
      end

      def save_file(content:)
        raise 'Invalid output file path' unless valid_output_file_extension?

        File.open(output_path, 'w') do |file|
          file.write(content)
        end
      end

      private

      def valid_input_file_extension?
        File.extname(input_path) == '.bin'
      end

      def valid_output_file_extension?
        File.extname(output_path) == '.txt'
      end
    end
  end
end

File: python/src/file_handler/decompress.py
from file_handler.base import BaseHandler
import os

class DecompressFileHandler(BaseHandler):
  def load(self) -> bytes:
    if not self._is_input_path_valid():
      raise Exception('Invalid File')
    
    with open(self.input_path, 'rb') as file:
      return file.read()
  
  def save_file(self, **kwargs) -> None:
    if not self._is_output_path_valid():
      raise Exception('Invalid output file path')
    
    with open(self.output_path, 'w') as file:
      file.write(kwargs.get('content'))
  
  def _input_file_extension_is_valid(self) -> bool:
    return os.path.splitext(self.input_path)[-1] == '.bin'

  def _output_file_extension_is_valid(self) -> bool:
    return os.path.splitext(self.output_path)[-1] == '.txt'